src/controllers/hsm/hsm.controller.js

```diff
@@ -94,10 +99,10 @@ async function startHsmWallet(req, res) {
   // Obtains this wallet's xPub from the HSM
   let xPub = null;
   try {
-    xPub = await hsmService.getXPubFromKey(connectionObj, hsmKeyName, {
-      isReadOnlyWallet: true,
-    });
+    await hsmService.deriveMainKeysFromRoot(connectionObj, hsmKeyName);
```

Comment:
Not at all, but the method was generating all temporary keys.
After some dicussion with the devs at dinamo they recommended that some derivations are created as not temporary keys, the most important being the  account and change derivation.

The one before would derive to account and stop, this one derives to change making them persistent on the HSM.
This method also checks the existence of the keys before deriving (since they are persistent) saving time on subsequent initializations.